plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/CheckpointMetadataEntry.java

```diff
@@ -14,20 +14,34 @@
 package io.trino.plugin.deltalake.transactionlog;
 
 import com.google.common.collect.ImmutableMap;
+import io.airlift.slice.SizeOf;
 
 import java.util.Map;
 import java.util.Optional;
 
 import static com.google.common.base.Preconditions.checkArgument;
+import static io.airlift.slice.SizeOf.SIZE_OF_LONG;
+import static io.airlift.slice.SizeOf.estimatedSizeOf;
+import static io.airlift.slice.SizeOf.instanceSize;
+import static io.airlift.slice.SizeOf.sizeOf;
 import static java.util.Objects.requireNonNull;
 
 // https://github.com/delta-io/delta/blob/master/PROTOCOL.md#checkpoint-metadata
 public record CheckpointMetadataEntry(long version, Optional<Map<String, String>> tags)
 {
+    private static final int INSTANCE_SIZE = instanceSize(CheckpointMetadataEntry.class);
+
     public CheckpointMetadataEntry
     {
         checkArgument(version > 0, "version is not positive: %s", version);
         requireNonNull(tags, "tags is null");
         tags = tags.map(ImmutableMap::copyOf);
     }
+
+    public long getRetainedSizeInBytes()
```

Comment:
I usually avoid `get` prefix in record. No requested change because some other entry classes are still non-record.